enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -6584,6 +7005,16 @@ class AdjointGenerator
       assert(gutils->invertedPointers.find(orig) ==
              gutils->invertedPointers.end());
 
+      if (Mode == DerivativeMode::ForwardMode) {
+        IRBuilder<> Builder2(&call);
```

Comment:
You need to check if !isConstantValue(orig->getArgOperand(0))

and also invertPointer

Reply:
Add a runtime check that inverted ptr != original ptr

Reply:
done 3a04ef2142b44e66876b68c2b559b660f7929533